packages/eslint-plugin-template/src/rules/no-duplicate-attributes.ts

```diff
@@ -0,0 +1,89 @@
+import { TmplAstElement } from '@angular/compiler';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+import {
+  BoundAttribute,
+  BoundEvent,
+  TextAttribute,
+} from '@angular/compiler/src/render3/r3_ast';
+
+type Options = [];
+export type MessageIds = 'noDuplicateAttributes';
+export const RULE_NAME = 'no-duplicate-attributes';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that no components have duplicate input properties or output event listeners',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      noDuplicateAttributes:
+        'The attribute "{{attributeName}}" was declared multiple times',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+
+    return {
+      Element({ inputs, outputs, attributes }: TmplAstElement) {
+        const duplicateInputsAndAttributes = findDuplicates([
+          ...inputs,
+          ...attributes,
+        ]);
+
+        const nonTwoWayBindingOutputs = outputs.filter((output) => {
+          return !inputs.find(
+            (input) =>
+              input.sourceSpan.start === output.sourceSpan.start &&
+              input.sourceSpan.end === output.sourceSpan.end,
+          );
+        });
+
+        const duplicateOutputs = findDuplicates(nonTwoWayBindingOutputs);
```

Comment:
By having 2 calls, my thinking was that it would be faster, as you can assume there won't be any inputs or outputs with the same name, if you merge the 2 arrays together you will have to iterate more elements when checking for duplicates. i.e. when checking for a duplicate input I would have to iterate over all of the outputs to check there isn't a duplicate which I know won't be the case.